### PR TITLE
Admission flow for create pod

### PR DESCRIPTION
**Pull request by anoipm**

### CODE REVIEW

File: internal/admission/defaulting_test.go [hunk 1]
```diff
@@ -220,52 +220,60 @@ func TestFlow(t *testing.T) {
...
 			require.NotNil(t, res)
 			require.True(t, res.AdmissionResponse.Allowed)
-			require.Equal(t, tt.want.patchesCount, len(res.Patches))
-			if tt.want.patchesCount > 0 {
-				patchValue := (res.Patches[0].Value).(map[string]interface{})
-				require.Contains(t, patchValue, pkg.PodValidationLabel)
-				require.Equal(t, tt.want.validationStatus, patchValue[pkg.PodValidationLabel])
+
+			if tt.shouldCallValidate {
+				mockImageValidator.AssertNumberOfCalls(t, "Validate", 1)
```
Comment (dbadura) [inline internal/admission/defaulting_test.go:225]: Can we replace it with `expectedPatch`?

Comment (anoipm) [reply internal/admission/defaulting_test.go:263]: done

File: internal/admission/defaulting_test.go [hunk 2]
```diff
@@ -126,92 +127,129 @@ func TestFlow(t *testing.T) {
...
-			name: "pod with label Failed should pass without validation and unchanged",
-			args: args{
-				labels:               map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusFailed},
-				shouldNotBeValidated: true,
+			name:        "update pod with label Failed should pass without validation",
+			operation:   admissionv1.Update,
+			inputLabels: map[string]string{pkg.PodValidationLabel: pkg.ValidationStatusFailed},
+			want: want{
+				shouldCallValidate: false,
+				patches:            []jsonpatch.JsonPatchOperation(nil),
```
Comment (dbadura) [inline internal/admission/defaulting_test.go:173]: is `[]jsonpatch.JsonPatchOperation{}`different?

File: internal/admission/defaulting_test.go (hunk 1 above)
Comment (dbadura) [reply internal/admission/defaulting_test.go:263]: Different response patches are produced by those 2 functions even when nothing to do
- admission.Allowed("validation is not needed for pod")
- admission.PatchResponseFromRaw(req.Object.Raw, fBytes)

File: internal/admission/defaulting_test.go (hunk 2 above)
Comment (dbadura) [reply internal/admission/defaulting_test.go:173]: Different response patches are produced by those 2 functions even when nothing to do
- admission.Allowed("validation is not needed for pod")
- admission.PatchResponseFromRaw(req.Object.Raw, fBytes)

**Review by dbadura — APPROVED** [review]

**Pull request merged**